### ISSUE TITLE
🐛 BUG: Inline HTML in markdown files

### ISSUE THREAD

**Issue by svemat01** [withastro/astro#2277]
## Workaround [For users with this issue]
A temporary workaround for this issue is the surround your markdown paragraph with a `<p>` tag. While this isn't the most good-looking solution, it works until we get an actual solution.

### What version of `astro` are you using?

0.22.1

### What package manager are you using?

yarn

### What operating system are you using?

Linux

### Describe the Bug

When writing Markdown files, Inline HTML is not properly supported. In the case that we would add an inline HTML tag to a markdown text block, the whole text block would be marked as HTML content, which by extension means that the block of text that before would have been turned into a `p` tag is now instead just text. This completely breaks our ability to style that text as sometimes it is a p tag, and other times it is just text. This behavior can also be observed using custom components, which in turn also work as HTML tags to some degree.

This issue has been brought up in a support thread in this project's Discord server. @sarah11918 took a big part in the discussion and recommended I report this error here. The discussion was held in the thread with the name `Svemat (2021-12-25)` with channel id `924430559999037491`.

In the attached link to a sandbox, it can be observed in the `mixed.md` that the text block that includes the `strong` tag produces an output where the text is plain HTML text while the text block without inline HTML produces an expected `p` tag with the text block content.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-yja7xp?devtoolsheight=33&file=src/pages/mixed.md

**Comment by jonathantneal** [comment]
Thanks for the example. I would have expected the text to be wrapped in a paragraph, either way, unless this is outlined differently in some official markdown specification. @natemoo-re?

**Example**: Currently wraps the 3rd line in a `<p>`:

```md
# title

lorem ipsum dolor sit amet.
```


**Example**: Currently woes not wrap the 3rd line in a `<p>`:

```md
# title

lorem ipsum <strong>dolor</strong> sit amet.
```

**Comment by FredKSchott** [comment]
Fixed in v0.24.0
